Add Rust 1.90.0 announcement (#1691)
rust-lang/blog.rust-lang.org: content/Rust-1.90.0.md

+++
path = "2025/09/18/Rust-1.90.0"
title = "Announcing Rust 1.90.0"
authors = ["The Rust Release Team"]
aliases = ["releases/1.90.0"]

[extra]
release = true
+++

The Rust team is happy to announce a new version of Rust, 1.90.0. Rust is a programming language empowering everyone to build reliable and efficient software.

If you have a previous version of Rust installed via `rustup`, you can get 1.90.0 with:

```console
$ rustup update stable
```

If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.90.0](https://doc.rust-lang.org/stable/releases.html#version-1900-2025-09-18).

If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!

## What's in 1.90.0 stable

### `lld` by default on `x86_64-unknown-linux-gnu`

> As [previously announced](https://blog.rust-lang.org/2025/09/01/rust-lld-on-1.90.0-stable/),

Author: Happy to take that text - we're not good at finding when people do actually write blurbs yet. Feel free to push or suggest it, otherwise I'll try to copy it in the next few days.

1.90 brings a change in the default linker to `lld` on
`x86_64-unknown-linux-gnu`. As discussed in that blog, for the vast majority of
users this change should just be a compile time reduction. In some edge cases,
`lld` may have different behavior: please file
[issues](https://github.com/rust-lang/rust/issues/new/choose) if you run into

> problems as part of the migration.

Reviewer: You can use a four backtick suggestion block and then use three backticks within this block.

In the vast majority of cases, LLD should be backwards compatible with BFD, and you should not see any difference other than reduced compilation time. However, if you do run into any new linker issues, you can always opt out using the `-C linker-features=-lld` compiler flag. Either by adding it to the usual `RUSTFLAGS` environment variable, or to a project's [`.cargo/config.toml`](https://doc.rust-lang.org/cargo/reference/config.html) configuration file,
like so:

```toml
[target.x86_64-unknown-linux-gnu]
rustflags = ["-Clinker-features=-lld"]
```

If you encounter any issues with the LLD linker, please [let us know](https://github.com/rust-lang/rust/issues/new/choose). You can read more about the switch to LLD, some benchmark numbers and the opt out mechanism [here](https://blog.rust-lang.org/2025/09/01/rust-lld-on-1.90.0-stable/).

### Cargo adds native support for workspace publishing

`cargo publish --workspace` is now supported, automatically publishing all of
the crates in a workspace in the right order (following any dependencies
between them). This has long been possible with external tooling or manual

> ordering of individual publishes, but this brings the functionality into Cargo

Reviewer 2: If we are comparing to existing approaches, is it worth it to call out how this improves on those?

The biggest change is in the verify step which can verify everything before publishing
- for a real publish, this reduces the chance of it failing in a partially published state
- for dry-run, you can now run the verify

> itself.

Reviewer 2: Should we call out the caveat that the publish is not atomic and network or server side errors can leave you in a partially published state?

This is true for existing workarounds but people might have a different expectation with it built in.

### Demoting `x86_64-apple-darwin` to Tier 2 with host tools

GitHub will soon [discontinue][gha-sunset] providing free macOS x86\_64 runners for public repositories. Apple has also announced their [plans][apple] for discontinuing support for the x86\_64 architecture.

> In accordance with these changes, as of Rust 1.90, the project has [demoted the `x86_64-apple-darwin` target][rfc] from [Tier 1 with host tools](https://doc.rust-lang.org/stable/rustc/platform-support.html#tier-1-with-host-tools) to [Tier 2 with host tools](https://doc.rust-lang.org/stable/rustc/platform-support.html#tier-2-with-host-tools). This means that the target, including tools like `rustc` and `cargo`, will be guaranteed to build but is not guaranteed to pass our automated test suite.

Reviewer 3: ```suggestion
In accordance with these changes, as of Rust 1.90, we have [demoted the `x86_64-apple-darwin` target][rfc] from [Tier 1 with host tools](https://doc.rust-lang.org/stable/rustc/platform-support.html#tier-1-with-host-tools) to [Tier 2 with host tools](https://doc.rust-lang.org/stable/rustc/platform-support.html#tier-2-with-host-tools). This means that the target, including tools like `rustc` and `cargo`, will be guaranteed to build but is not guaranteed to pass our automated test suite.
```
Not sure if people know what "the project" is here.
